netwaystev2/src/transport/transport.rs

```diff
@@ -0,0 +1,231 @@
+use super::endpoint::EndpointData;
+use super::interface::{
+    Packet,
+    TransportCmd::{self, *},
+    TransportNotice, TransportQueueKind, TransportRsp,
+};
+use super::udp_codec::{LinesCodec, NetwaystePacketCodec};
+use crate::common::Endpoint;
+use crate::settings::*;
+
+use std::time::Duration;
+use std::{net::SocketAddr, pin::Pin};
+
+use anyhow::Result;
+use futures::prelude::*;
+use futures::stream::Fuse;
+use futures::StreamExt;
+use stream::{SplitSink, SplitStream};
+use tokio::net::UdpSocket;
+use tokio::sync::mpsc::{self, Receiver, Sender};
+use tokio::time as TokioTime;
+use tokio_stream::wrappers::IntervalStream;
+use tokio_util::udp::UdpFramed;
+
+type TransportCmdSend = Sender<TransportCmd>;
+type TransportCmdRecv = Receiver<TransportCmd>;
+
+type TransportRspSend = Sender<TransportRsp>;
+type TransportRspRecv = Receiver<TransportRsp>;
+
+type TransportNotifySend = Sender<TransportNotice>;
+type TransportNotifyRecv = Receiver<TransportNotice>;
+
+type TransportInit = (Transport, TransportCmdSend, TransportRspRecv, TransportNotifyRecv);
+
+//type TransportItem = (Packet, SocketAddr);
+type TransportItem = (String, SocketAddr);
+
+pub struct Transport {
+    requests:        TransportCmdRecv,
+    responses:       TransportRspSend,
+    notifications:   TransportNotifySend,
+    //udp_stream_send: SplitSink<UdpFramed<NetwaystePacketCodec>, (Packet, SocketAddr)>,
+    //udp_stream_recv: Fuse<SplitStream<UdpFramed<NetwaystePacketCodec>>>,
+    udp_stream_send: SplitSink<UdpFramed<LinesCodec>, TransportItem>,
+    udp_stream_recv: Fuse<SplitStream<UdpFramed<LinesCodec>>>,
+
+    //endpoints: EndpointData<PacketInfo>,
+    endpoints: EndpointData<String>,
+}
+
+impl Transport {
+    pub fn new(opt_host: Option<&str>, opt_port: Option<u16>) -> Result<TransportInit> {
+        // Bind socket to UDP
+        let udp_socket = bind(opt_host, opt_port)?;
+
+        // Split the socket into a two-part stream
+        let udp_stream = UdpFramed::new(udp_socket, LinesCodec::new());
+        let (udp_stream_send, udp_stream_recv) = udp_stream.split();
+        let udp_stream_recv = udp_stream_recv.fuse();
+
+        // Build the Transport
+        let (cmd_tx, cmd_rx): (TransportCmdSend, TransportCmdRecv) = mpsc::channel(TRANSPORT_CHANNEL_LEN);
+        let (rsp_tx, rsp_rx): (TransportRspSend, TransportRspRecv) = mpsc::channel(TRANSPORT_CHANNEL_LEN);
+        let (notice_tx, notice_rx): (TransportNotifySend, TransportNotifyRecv) = mpsc::channel(TRANSPORT_CHANNEL_LEN);
+
+        Ok((
+            Transport {
+                requests: cmd_rx,
+                responses: rsp_tx,
+                notifications: notice_tx,
+                udp_stream_send,
+                udp_stream_recv,
+                endpoints: EndpointData::new(),
+            },
+            cmd_tx,
+            rsp_rx,
+            notice_rx,
+        ))
+    }
+
+    pub async fn monitor(&mut self) -> Result<()> {
```

Comment:
Can we name this something else, like `run`, since it's doing more than just monitoring. I would expect something that "monitors" to be a read-only thing...